hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/RMStateStore.java

```diff
@@ -1253,7 +1255,9 @@ protected void handleStoreEvent(RMStateStoreEvent event) {
 
       final RMStateStoreState oldState = getRMStateStoreState();
 
+      long startTime = clock.getTime();
       this.stateMachine.doTransition(event.getType(), event);
+      this.metrics.increment(event.getType(), clock.getTime() - startTime);
```

Comment:
This code will be called twice for the events propagated with the dispatcher (once in the AsyncDispatcher class and once here). We should remove dispatcher.addMetrics(metrics, metrics.getEnumClass());

Reply:
you're right, updated